packages/material-ui/src/styles/createMuiTheme.d.ts

```diff
@@ -13,7 +13,7 @@ import { ComponentsProps } from './props';
 export type Direction = 'ltr' | 'rtl';
 
 export interface ThemeOptions {
-  shape?: ShapeOptions;
+  shape?: Shape;
```

Comment:
Don't we still need the partial?

Reply:
Unlike `spacing`, which is `deepmerge`d, `shape` is not.
https://github.com/mui-org/material-ui/blob/ca33d9773581a8b0ff37c92260b48c9f72b42de9/packages/material-ui/src/styles/createMuiTheme.js#L38-L44

https://github.com/mui-org/material-ui/blob/ca33d9773581a8b0ff37c92260b48c9f72b42de9/packages/material-ui/src/styles/createMuiTheme.js#L35

This means that for `spacing`, we can specify just partial of it, but for `shadow`, we need to specify the full props, from what I understand.

Reply:
FYI - if you take a look at `shadow` type definition, it is not `Partial<>` either.
https://github.com/mui-org/material-ui/blob/ca33d9773581a8b0ff37c92260b48c9f72b42de9/packages/material-ui/src/styles/createMuiTheme.d.ts#L23

Reply:
Thanks for raising the issue! Should we fix the implementation over the TypeScript definition then? I have missed this part. I think that spacing and shape should behave the same way.

Reply:
I see. I'll do this. BTW, what about `shadows`? Should it also be a partial in the `ThemeOptions`?

Reply:
> we need to specify the full props

I don't think that it's the behavior people expect. We need to fix it :)

Reply:
NVM. `shadows` is different.

Reply:
Regarding the shadows, yeah it's fine. Relative values are important, people need to provide the full property.